### ISSUE TITLE
Unknown intrinsic gather

### ISSUE THREAD

**Issue by mym2009** [halide/Halide#6776]
hi, I use halide 12.0.1 to compile test/correctness/gather.cpp. When generate target:arm-64-android-hvx_128-hvx_v65, generator throw an error:
Unhandled exception: Internal Error at /home/ding/Downloads/Halide-12.0.1/src/CodeGen_LLVM.cpp:3171 triggered by user code at : Unknown intrinsic gather.  But remove hvx_v65 from target, it works. I see vgather has been supported in version 12.0.1.
Who can tell me the reason? Thanks.

**Comment by pranavb-ca** [comment]
The reason this passes for you when running without hvx_v65 is becasue the gather instruction isnt generated if no hvx_v65 is used.

**Comment by mym2009** [comment]
Yes, I know only v65+ hexagon dsp support vgather. I want to know the reason why target hvx_v65 cannot generate vgather code.

**Comment by abadams** [comment]
> [!NOTE]
> **Automated triage:** this issue was flagged by Claude as likely safe to close, for the reason below. The reasoning was checked against the current source, and a second pass tried and failed to refute it, but no human verified this specific text. **If the reason is wrong, please reopen the issue and say so** — some fraction of these are expected to be wrong.

**Reason (CLOSE_STALE):** Reported against Halide 12.0.1 with the LLVM of that era, neither of which is supported (Halide now requires LLVM 21-23). The HVX gather intrinsic is `Call::hvx_gather` and is handled in `CodeGen_Hexagon.cpp`; the same VTCM gather pipeline compiles all the way through LLVM codegen for `arm-64-android-hvx-hvx_v65` on current main without error. The reporter never followed up on a supported version.

**Evidence:** src/CodeGen_Hexagon.cpp:2013-2019 handles `Call::hvx_gather`; src/CodeGen_Hexagon.cpp:483-486 runs `scatter_gather_generator` for v65+. Rebuilt the `test/correctness/gather.cpp` pipeline and ran `compile_to_object` for `arm-64-android-hvx-hvx_v65` and `hexagon-32-noos-hvx-hvx_v65` on current main for uint8/uint16/int32: all succeed.

Closing: this was filed against Halide 12.0.1, which is far outside the supported LLVM/Halide range today, and the same VTCM gather pipeline compiles cleanly for `arm-64-android-hvx-hvx_v65` on current main. Please open a fresh issue with a self-contained repro if you still hit this on a recent release.
